### PR TITLE
Add recipe for Jupyter Scheduler

### PR DESCRIPTION
**Pull request by andrii-i**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/jupyter-scheduler/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7
+    - hatchling >=1.3.1
+    - jupyterlab >=4.0,<5.dev0
+    - pip
```
Comment (3coins) [inline recipes/jupyter-scheduler/meta.yaml:22]:
```suggestion
    - pip
    - hatch-jupyter-builder
```

**Comment by JasonWeill** [comment]
I agree to be a maintainer for jupyter-scheduler.

**Comment by 3coins** [comment]
I agree to be a maintainer for jupyter-scheduler.

**Comment by dlqqq** [comment]
I agree to be a maintainer.

**Comment by andrii-i** [comment]
@conda-forge/staged-recipes this PR is ready for review, please give it a look when possible. 

Recipe adds a JupyterLab extension for running notebook jobs. This extension is composed of a Python package named jupyter_scheduler for the server extension and a TS package named @jupyterlab/scheduler for the frontend extension. Documentation is available on [ReadTheDocs](https://jupyter-scheduler.readthedocs.io/).

**Comment by andrii-i** [comment]
Tagging @conda-forge/help-python, @conda-forge/help-nodejs, @conda-forge/staged-recipes to potentially route this PR (ready for review) to relevant teams

**Comment by andrii-i** [comment]
@conda-forge-admin, please ping team. PR is ready for review.

**Comment by JasonWeill** [comment]
@conda-forge/help-python Ready for review.

**Comment by andrii-i** [comment]
@conda-forge/help-python Ready for review.

File: recipes/jupyter-scheduler/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - jupyter_server >=1.6,<3
+    - traitlets >=5.0,<6.dev0
+    - nbconvert >=7.0,<8.dev0
+    - pydantic >=1.10,<2.dev0
+    - sqlalchemy >=1.0,<2.dev0
+    - croniter >=1.4,<2.dev0
+    - pytz ==2023.3
+    - fsspec ==2023.6.0
+    - s3fs ==2023.6.0
+    - psutil >=5.9,<6.dev0
```
[the remark this answers is not in the document]
Comment (andrii-i) [reply recipes/jupyter-scheduler/meta.yaml]: Good point, thank you

[the remark this answers is not in the document]
Comment (andrii-i) [reply recipes/jupyter-scheduler/meta.yaml]: Excluded pre-release from version spec upper bound

**Pull request merged**